libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/MapboxNavigationApp.kt

```diff
@@ -131,6 +140,15 @@ object MapboxNavigationApp {
         mapboxNavigationAppDelegate.unregisterObserver(mapboxNavigationObserver)
```

Comment:
Out of date comment here, but making a note. Was attempting to build this type of function

``` kotlin
   fun <T : MapboxNavigationObserver> registerObserver(lifecycleOwner: LifecycleOwner, observer: MapboxNavigationObserver) {
        return mapboxNavigationAppDelegate.getObserver(kClass)
    }
```

It turned out to be complicated. Maintaining a map of observers, checking on every event if it is supposed to register or unregister the observer. 

There is also an issue with the interface. It's like it should be
``` kotlin
    fun <T : MapboxNavigationObserver> registerObserver(lifecycleOwner: LifecycleOwner, kClass: KClass<T>) {
        return mapboxNavigationAppDelegate.getObserver(kClass)
    }
```

But then we also need to create a factory provider. So I'm finding this solution to be a bit complicated